neuron.schema.json

```diff
@@ -0,0 +1,148 @@
+{
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "$id": "https://raw.githubusercontent.com/GenGeNN/GenGeNN-resources/master/neuron.json.schema",
+    "title": "Neuron",
+    "description": "A single neuron definition for GeNN JSON",
+    "type": "object",
+    "properties": {
+        "friendlyName": {
+            "description": "A friendly name for the neuron class",
+            "type": "string"
+        },
+        "summaryDescription": {
+            "description": "A description of the function of the neuron",
+            "type": "string"
+        },
+        "className": {
+            "description": "The programmatic name for the class",
+            "type": "string"
+        },
+        "parameters": {
+            "description": "All the parameter names for the simulation",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "variables": {
+            "description": "All the variable names and types for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
```

Comment:
I *think* these can be all scalar types, including "scalar" but also explicit "float", "double", "int" (and other C integer types). I am not aware of "vector" a s possible variable type here.

Reply:
whoops, just assumed that without thinking - I'll correct in new commit

Reply:
yeah, all c types (including uint8_t, uint16_t, uint32_t, uint64_t etc) and scalar are what is generally supported here

Reply:
fair enough - I'll remove validation for now and we could add regex if needed at a later date...